### PR TITLE
Feat add sample workflow documentation

### PR DESCRIPTION
**Pull request by thsetz**
- doc-source/sample_workflow.rst has been rewritten to use doctests
- fossology/*.py files with doctest entries have been rewritten so that tests are executed.
some line-tests are skipped - needs discussion on which could be enabled with appropriate effort.

- a pytest.ini was created adding. pytest opts, so that doctests are executed whenever pytest is called. 
Whenever pytest>=6 is used that file can be deleted and the "addopt entry" can migrate to pyproject.toml

Tests pass till they call "poetry run codecov -t ${{ secrets.CODECOV_TOKEN }}" in fossologytests.yml. The secret is not defined in my environment - that is why i did the pull request. 
Maybe this call should only be executed if it is executed in the repository "fossology/fossology-python"

### CODE REVIEW

File: docs-source/sample_workflow.rst [hunk 1]
```diff
@@ -15,118 +15,109 @@ Sample Workflow
...
-    print("token written to .token")
-  else:
-    print("Reuse token from .token")
-    # Load the token
-    with open(".token", "r") as fp:
-        token = fp.read()
+
+>>> path_to_token_file = pathlib.Path.cwd() / '.token'
+>>> if not path_to_token_file.exists():
+...   #print("Enter your Fossology credentials, e.g. in the test environment 'username: fossy' and 'password: fossy'")
```
Comment (deveaud-m) [inline docs-source/sample_workflow.rst:37]: Adding some comments here to explain why the `input()` method is not called here could be helpful.

File: fossology/uploads.py [hunk 2]
```diff
@@ -36,10 +35,10 @@ def detail_upload(
         :Examples:
 
         >>> # Wait up to 20 minutes until the upload is ready
-        >>> long_upload = detail_upload(1, wait_time=120)
+        >>> long_upload = u.detail_upload(1, wait_time=120) # doctest: +SKIP
```
Comment (deveaud-m) [inline fossology/uploads.py:38]:
```suggestion
        >>> long_upload = detail_upload(1, wait_time=120) # doctest: +SKIP
```

**Review by deveaud-m — CHANGES_REQUESTED** [review]

**Comment by deveaud-m** [comment]
Working like a charm in my environment.

I just added some minor improvement suggestions.

Maybe we could use this patch to switch to pytest 6+?

File: fossology/uploads.py (hunk 2 above)
Comment (thsetz) [reply fossology/uploads.py:38]: fixed typo / leftover from test

BTW: I skipped (# doctest: +SKIP) some doctests in fossology/*.py. The current doctest environment has no "global scope"  saying each doctest block runs in it's own environment. To enable more doctests (remove those # doctest: +SKIP suffixes to a doctest line) some rework has to be done.

File: docs-source/sample_workflow.rst (hunk 1 above)
Comment (deveaud-m) [reply docs-source/sample_workflow.rst]: @thsetz please resolve the discussion when you commit new changes.

**Review by deveaud-m — APPROVED** [review]

**Comment by deveaud-m** [comment]
This looks good now. Merging.

**Pull request merged**